giturlparse/parser.py

```diff
@@ -70,29 +70,30 @@ def parse(self):
             'owner': None,
         }
 
-        regexp = (r'^(https?|git|ssh|rsync)\://'
-                  '(?:(.+)@)*'
-                  '([a-z0-9_.-]*)'
-                  '[:/]*'
-                  '([\d]+){0,1}'
-                  '(/(.+)/(.+).git)')
-        m1 = re.search(regexp, self._url)
-
-        regexp = (r'^(?:(.+)@)*'
-                  '([a-z0-9_.-]*)[:/]*'
-                  '([\d]+){0,1}'
-                  '([:/](.+)/(.+).git)')
-        m2 = re.search(regexp, self._url)
-
-        if m1:
+        regexes = [
+            (r'^(https?|git|ssh|rsync)\://'
+                '(?:(.+)@)*'
+                '([a-z0-9_.-]*)'
+                '[:/]*'
+                '([\d]+){0,1}'
+                '(/(.+)/(.+).git)'),
+            (r'^(?:(.+)@)*'
+                '([a-z0-9_.-]*)[:/]*'
+                '([\d]+){0,1}'
+                '([:/](.+)/(.+).git)')
+        ]
+
+        if re.search(regexes[0], self._url):
```

Comment:
the reason to have an array is so that you can do looping.

you'll need to define capture groups in the two regexes for this to work.

then you have one problem: `re.sub(r'^:', '', m2.group(4))` needs to be converted to a single capture group.

Reply:
@jayvdb, I am not getting how we are going to do looping. Actually, capture groups in each of the regexes are different. 
>you'll need to define capture groups in the two regexes for this to work.

Can you explain this line a bit more?
Also, I am thinking of replacing m2 with m4 in my other PR which matches all URLs in m2.
So, maybe we don't have to solve `re.sub(r'^:', '', m2.group(4))` problem?